src/libraries/System.Text.Json/tests/Serialization/Value.ReadTests.GenericCollections.cs

```diff
@@ -1161,5 +1179,41 @@ public static void Read_HigherOrderCollectionInheritance_Works()
             Assert.Equal("test", JsonSerializer.Deserialize<MyMyList<string>>(json).First());
             Assert.Equal("test", JsonSerializer.Deserialize<MyListString>(json).First());
         }
+
+        [Fact]
+        public static void Read_Generic_NoPublicConstructor_Throws()
```

Comment:
We should be able to use `[Theory]` + the non-generic `JsonSerializer.Deserialize` method here and elsewhere.

Reply:
Fixed in bccf92d

In a few tests, we reference data from fields of the form `public static readonly string s_json`, which cannot be used directly by `[InlineData]`, so I refactored to `[MemberData]` tests in 
921fc3a.  If we don't like the way `[MemberData]` works out for those tests, I can roll that back.

Reply:
This change is fine.